images/hub/unfrozen/requirements.txt

```diff
@@ -0,0 +1,33 @@
+# This file is the input to requirements.txt, which is a frozen version of this.
+#
+# To update:
+# - the jupyterhub version or the frozen requirements.txt file, use the
+#   "Run workflow" button at https://github.com/jupyterhub/zero-to-jupyterhub-k8s/actions/workflows/watch-dependencies.yaml.
+# - the major version pins in this file, trigger dependabot to run via
+#   https://github.com/jupyterhub/zero-to-jupyterhub-k8s/network/updates
+#
+
+# JupyterHub itself
+jupyterhub==5.2.1
+
+# JupyterHub Spawner, kubernetes specific
+jupyterhub-kubespawner==7.*
+
+# JupyterHub Authenticator choices
+jupyterhub-firstuseauthenticator==1.*
+jupyterhub-hmacauthenticator==1.*
+jupyterhub-ldapauthenticator==2.*
+jupyterhub-ltiauthenticator==1.*
+jupyterhub-nativeauthenticator==1.*
+jupyterhub-tmpauthenticator==1.*
+oauthenticator[googlegroups,mediawiki]==17.*
+
+# JupyterHub service shutting servers after a period of inactivity
+jupyterhub-idle-culler==1.*
+
+# Other optional dependencies for additional features
+pymysql==1.*  # mysql
+psycopg2==2.*  # postgres
+pycurl==7.*  # internal http requests handle more load with pycurl
+sqlalchemy-cockroachdb==2.* # cocroachdb
+statsd==4.*  # statsd metrics collection (TODO: remove soon, since folks use prometheus)
```

Comment:
I think we should treat most transient dependencies that are internal to JupyterHub as an implementation detail, unless our documentation assumes they've available for extension.

Potentially we could treat `pymysql` `psycopg2` as possibly breaking since they affect the integration of Z2JH with an external database (e.g. a new release of these libraries could prevent Z2JH being upgraded if someone's running a very old DB), but given that this isn't something we've considered before, and in practice it's rare to have breaking changes in database APIs, we can make maintenance easier by ignoring it.